aspnet/3-binary-data/Controllers/BooksController.cs

```diff
@@ -127,12 +127,18 @@ public ActionResult Edit(long? id)
         // POST: Books/Edit/5
         [HttpPost]
         [ValidateAntiForgeryToken]
-        public ActionResult Edit(Book book, long id)
+        public async Task<ActionResult> Edit(Book book, long id, HttpPostedFileBase image)
         {
             if (ModelState.IsValid)
             {
                 book.Id = id;
                 _store.Update(book);

```

Comment:
Done.  Plus, a little more logic was required to handle the case when the book already has an image, and the user just edited the text.